connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -209,6 +212,39 @@ private void scanUrlsAndAddPlugins(
             addPlugins(plugins.transformations(), loader);
             transformations.addAll(plugins.transformations());
         }
+
+        loadJdbcDrivers(loader);
+    }
+
+    private void loadJdbcDrivers(final ClassLoader loader) {
+        // Apply here what java.sql.DriverManager does to discover and register classes
+        // implementing the java.sql.Driver interface.
+        AccessController.doPrivileged(
+                new PrivilegedAction<Void>() {
+                    public Void run() {
+                        ServiceLoader<Driver> loadedDrivers = ServiceLoader.load(
```

Comment:
huh, i didn't know this existed. as a side note, should we consider using this instead of the reflections library? and do we know if there are any functional/performance differences?

Reply:
I would love it if we could do this, as it would dramatically improve startup times. It would require each connector adding a META-INF file to their JARs, which is feasible but not ideal.

Reply:
I had checked `ServiceLoader` when starting to implement classloading isolation, but because it required such metadata, it didn't come up as a general solution then and I moved on with reflections. 

Given a general packaging process for plugins and the fact that reflections take time (proportional to the number of dependencies as well as the number of plugin types) we could reevaluate this and require plugin packages to list their plugin classes under `META-INF`.

Right now, `ServiceLoader` won't work OOTB for Connect plugins.